### PR TITLE
Basic support for `dynamo/package`

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
This adds basic support for the `dynamo/package`. There remain additional features to be added to this container but I wanted to get the layout changes in as one PR and introduce other features one by one.

### Changes to `Card` props

`headlineSize: huge`
I added this prop value to support the extra large font size used on dynamos

`transparent: boolean`
We set background colour using `decidePalette` but the main image in a dynamo uses a transparent background so we need an imperative way to set this

`containerType`
We now pass the type of container down as the use case for the Dynamo is to vary some styles based on the type of container they are in. In this case, the variation is with the height of the top border line which shows for `dynamo/package`. Instead of 2 pixels we use 4.

| Frontend      | DCR      |
|-------------|------------|
| Screenshot 2022-06-24 at 11 10 06 | Screenshot 2022-06-24 at 11 09 44 |

## What's remaining?
[Floating sublinks](https://github.com/guardian/dotcom-rendering/issues/5267)
image

[Latest links](https://github.com/guardian/dotcom-rendering/issues/5196)
image

[Image Carousel](https://github.com/guardian/dotcom-rendering/issues/4612)

### CODE REVIEW

**Comment by HarryFischer** [comment]
🤩

File: dotcom-rendering/src/web/components/CardHeadline.tsx [hunk 1]
```diff
@@ -24,36 +25,50 @@ type Props = {
 	linkTo?: string; // If provided, the headline is wrapped in a link
 };
 
-const fontStyles = (size: SmallHeadlineSize) => {
+const fontStyles = (
+	size: SmallHeadlineSize,
+	containerPalette?: DCRContainerPalette,
+) => {
+	const options: FontScaleArgs = {};
+	if (containerPalette) options.fontWeight = 'bold';
```
Comment (OllysCoding) [inline dotcom-rendering/src/web/components/CardHeadline.tsx:33]: If this logic is just looking for container palette to be defined, would it be cleaner if this function took a parameter like `bold: boolean` and we did `fontSyles(size, !!containerPalette)` when calling it?

**Review by OllysCoding — APPROVED** [review]
+1 A really good start on this container :)

File: dotcom-rendering/src/web/components/CardHeadline.tsx (hunk 1 above)
Comment (OllysCoding) [reply dotcom-rendering/src/web/components/CardHeadline.tsx]: Very tidy! Nice 👍

**Comment by oliverlloyd** [comment]
Closes #5143

**Pull request merged**